types/errors/errors.go

```diff
@@ -394,3 +397,36 @@ type causer interface {
 type unpacker interface {
 	Unpack() []error
 }
+
+func GRPCWrap(err error, c codes.Code, msg string) error {
+	if err != nil {
+		st := status.New(c, msg)
+		var sdkErr *Error
+		if errors.As(err, &sdkErr) {
+			errorInfo := &errdetails.ErrorInfo{
+				Reason:   sdkErr.Error(),
+				Metadata: map[string]string{"Codespace": sdkErr.Codespace(), "ABCICode": fmt.Sprintf("%d", sdkErr.ABCICode())},
+			}
+			st, err = st.WithDetails(errorInfo)
+			if err != nil {
+				return err
+			}
+		}
+
+		stack := stackTrace(err)
+		var stackEntries []string
+		stackEntries = make([]string, len(stack))
+		for i, f := range stack {
+			stackEntries[i] = fmt.Sprintf("%+v", f)
+		}
+		dbgInfo := &errdetails.DebugInfo{
+			StackEntries: stackEntries,
+		}
+		st, err = st.WithDetails(dbgInfo)
+		if err != nil {
+			return err
+		}
```

Comment:
I would say here we should return the error received too, not this one 🤔 
```suggestion
		st, withDetailsErr = st.WithDetails(dbgInfo)
		if withDetailsErr != nil {
			return st.Err()
		}
```

Reply:
or simply log and avoid adding detail

Reply:
I removed stack trace, I think it has no much sense to returning to client, I think it's better to log it.